File: src/main/python/wordcount.py
import sys
from pyspark import SparkContext
from pyspark import SparkConf
"""
词频统计
"""
if __name__ == '__main__':
    if len(sys.argv) !=2 :
        print("统计wordcount <input>",file=sys.stderr)
        sys.exit(-1)

    conf = SparkConf().setMaster("local[2]").setAppName("spark03")
    sc = SparkContext(conf=conf)

    def printResult():
        counts = sc.textFile(sys.argv[1]).flatMap(lambda line:line.split("\t")).map(lambda x:(x,1)).reduceByKey(lambda a,b:a+b)
        output = counts.collect()

        for(word , count) in output:
            print("%s : %i" % (word,count))

    printResult()
    sc.stop()

File: src/main/scala/WordCount.scala
import org.apache.spark.{SparkConf, SparkContext}

object WordCount {
  def main(args:Array[String]): Unit ={
    val conf=new SparkConf().setAppName("wordcount").setMaster("local[*]")
    val sc=new SparkContext(conf)
    System.setProperty("HADOOP_USER_NAME", "root")

    val lines=sc.textFile("hdfs://master-ubuntu:9000/wordcount/wordcount.txt")

    val flatmaprdd=lines.flatMap(line=>line.split(" "))
    val maprdd=flatmaprdd.map(word=>(word,1))
    val reducerdd=maprdd.reduceByKey{case(x,y)=>x+y}

    reducerdd.saveAsTextFile("hdfs://master-ubuntu:9000/wordcount/output")
  }
}
